planner/core/hints.go

```diff
@@ -154,3 +165,121 @@ func (p *BlockHintProcessor) getCurrentStmtHints(hints []*ast.TableOptimizerHint
 	}
 	return p.QbHints[currentOffset]
 }
+
+func restoreOptimizerHint(hint *ast.TableOptimizerHint) string {
+	var sb strings.Builder
+	ctx := format.NewRestoreCtx(format.DefaultRestoreFlags, &sb)
+	err := hint.Restore(ctx)
+	// There won't be any error for optimizer hint.
+	if err != nil {
+		logutil.BgLogger().Warn("restore hint failed", zap.Error(err))
+	}
+	return sb.String()
+}
+
+// GenHintsFromPhysicalPlan generates hints from physical plan.
+func GenHintsFromPhysicalPlan(p Plan) string {
+	var hints []*ast.TableOptimizerHint
+	switch pp := p.(type) {
+	case *Update:
+		hints = genHintsFromPhysicalPlan(pp.SelectPlan, typeUpdate)
+	case *Delete:
+		hints = genHintsFromPhysicalPlan(pp.SelectPlan, typeDelete)
+	case PhysicalPlan:
+		hints = genHintsFromPhysicalPlan(pp, typeSelect)
+	}
+	hintsStr := make([]string, 0, len(hints))
+	for _, hint := range hints {
+		hintsStr = append(hintsStr, restoreOptimizerHint(hint))
+	}
+	return strings.Join(hintsStr, ", ")
+}
+
+func generateQBName(nodeType nodeType, offset int) model.CIStr {
+	if nodeType == typeDelete && offset == 0 {
+		return model.NewCIStr(defaultDeleteBlockName)
+	} else if nodeType == typeUpdate && offset == 0 {
+		return model.NewCIStr(defaultUpdateBlockName)
+	}
+	return model.NewCIStr(fmt.Sprintf("%s%d", defaultSelectBlockPrefix, offset))
```

Comment:
Can we use the same the way to generate all the kinds of query block names? For example:
```go
return model.NewCIStr(fmt.Sprintf("%s%d", xxxBlockPrefix, offset))
```

Reply:
Seems there is no easy way for `update` or `delete`, it may requires different type of prefix when there is a `select` stmt.